src/parser/mod.rs

```diff
@@ -6422,6 +6431,291 @@ impl<'a> Parser<'a> {
+                        self.expect_token(&Token::LParen)?;
+                        let op = self.parse_operator_name()?;
+                        self.expect_token(&Token::RParen)?;
+                        op
+                    } else {
+                        self.parse_operator_name()?
+                    };
+                    negator = Some(op_name);
+                }
+                Keyword::RESTRICT => {
+                    self.expect_token(&Token::Eq)?;
+                    let func_name = self.parse_object_name(false)?;
+                    restrict = Some(func_name);
+                }
+                Keyword::JOIN => {
+                    self.expect_token(&Token::Eq)?;
+                    let func_name = self.parse_object_name(false)?;
+                    join = Some(func_name);
+                }
+                _ => unreachable!("unexpected keyword in CREATE OPERATOR"),
+            }
+
+            // Check for comma or closing parenthesis
+            if !self.consume_token(&Token::Comma) {
+                break;
+            }
+        }
+
+        // Expect closing parenthesis
+        self.expect_token(&Token::RParen)?;
+
+        // FUNCTION is required
+        let function = function.ok_or_else(|| {
+            ParserError::ParserError("CREATE OPERATOR requires FUNCTION parameter".to_string())
+        })?;
+
+        Ok(Statement::CreateOperator(CreateOperator {
+            name,
+            function,
+            is_procedure,
+            left_arg,
+            right_arg,
+            commutator,
+            negator,
+            restrict,
+            join,
+            hashes,
+            merges,
+        }))
+    }
+
+    /// Parse a CREATE OPERATOR FAMILY statement
+    ///
+    /// [PostgreSQL Documentation](https://www.postgresql.org/docs/current/sql-createopfamily.html)
+    pub fn parse_create_operator_family(&mut self) -> Result<Statement, ParserError> {
+        let name = self.parse_object_name(false)?;
+        self.expect_keyword(Keyword::USING)?;
+        let using = self.parse_identifier()?;
+
+        Ok(Statement::CreateOperatorFamily(CreateOperatorFamily {
+            name,
+            using,
+        }))
+    }
+
+    /// Parse a CREATE OPERATOR CLASS statement
+    ///
+    /// [PostgreSQL Documentation](https://www.postgresql.org/docs/current/sql-createopclass.html)
+    pub fn parse_create_operator_class(&mut self) -> Result<Statement, ParserError> {
+        let name = self.parse_object_name(false)?;
+        let default = self.parse_keyword(Keyword::DEFAULT);
+        self.expect_keywords(&[Keyword::FOR, Keyword::TYPE])?;
+        let for_type = self.parse_data_type()?;
+        self.expect_keyword(Keyword::USING)?;
+        let using = self.parse_identifier()?;
+
+        let family = if self.parse_keyword(Keyword::FAMILY) {
+            Some(self.parse_object_name(false)?)
+        } else {
+            None
```

Comment:
Can we add a test case for a statement without a family?

Reply:
Done